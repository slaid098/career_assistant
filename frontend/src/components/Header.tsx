import React from 'react';
import Link from 'next/link';

const Header: React.FC = () => {
  return (
    <header className="bg-gray-800 text-white p-4 shadow-md">
      <div className="container mx-auto flex justify-between items-center">
        <Link href="/" className="text-2xl font-bold">
          Career Assistant
        </Link>
        <nav>
          {/* Future navigation links can go here */}
        </nav>
      </div>
    </header>
  );
};

export default Header; 