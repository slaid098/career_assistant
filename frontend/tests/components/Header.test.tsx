import React from 'react';
import { render, screen } from '@testing-library/react';
import Header from '@/components/Header';
import '@testing-library/jest-dom';

describe('Header Component', () => {
  it('should render the header with the site title', () => {
    render(<Header />);
    const titleElement = screen.getByText('Career Assistant');
    expect(titleElement).toBeInTheDocument();
  });

  it('should have a link to the homepage', () => {
    render(<Header />);
    const linkElement = screen.getByRole('link', { name: 'Career Assistant' });
    expect(linkElement).toHaveAttribute('href', '/');
  });

  it('should match the snapshot', () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });
}); 