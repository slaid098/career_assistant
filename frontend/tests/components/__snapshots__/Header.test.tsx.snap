// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Header Component should match the snapshot 1`] = `
<div>
  <header
    class="bg-gray-800 text-white p-4 shadow-md"
  >
    <div
      class="container mx-auto flex justify-between items-center"
    >
      <a
        class="text-2xl font-bold"
        href="/"
      >
        Career Assistant
      </a>
      <nav />
    </div>
  </header>
</div>
`;
